modules/media_duplicate_validation/src/Plugin/MediaDuplicateValidation/ColorMean.php

```diff
@@ -0,0 +1,382 @@
+   *
+   * @return float|int
+   *   Percent of the image that is deemed similar.
+   */
+  protected function getLikeness($image_one, $image_two) {
+    $image_one_colors = $this->getColorData($image_one);
+    $image_two_colors = $this->getColorData($image_two);
+
+    $different_pixels = 0;
+
+    // Go pixel by pixel and find how many different pixels the two images have.
+    foreach ($image_one_colors as $row_number => $row) {
+      foreach ($row as $column_number => $color_value) {
+
+        // Each pixel can be between 0 and 255. Use 256 to include 0 as a
+        // possible value.
+        $difference = abs($color_value - $image_two_colors[$row_number][$column_number]) / 256;
+
+        // Percent difference is greater than the allowed tollerance, its
+        // different.
+        if ((100 * $difference) > self::PIXEL_TOLLERANCE) {
+          $different_pixels++;
+        }
+      }
+    }
+
+    // Calculate the percent of the image that is different.
+    $total_pixels = pow(self::RESIZE_DIMENSION, 2);
+    $similarity = 100 * (($total_pixels - $different_pixels) / $total_pixels);
+    return $similarity;
+  }
+
+  /**
+   * Get a subset of all the images based on the column and row data.
+   *
+   * @param \Drupal\media\MediaInterface $entity
+   *   Media entity we are comparing.
+   * @param array $color_data
+   *   Gray scale multi-dimension array of pixel information.
+   *
+   * @return array
+   *   Array of media ids that are within the tolerance.
+   */
+  protected function getCloseMedia(MediaInterface $entity, array $color_data) {
+    $averages = $this->getRowColumnAverages($color_data);
+    $query = $this->database->select(self::DATABASE_TABLE, 't')
+      ->fields('t', ['mid']);
+
+    // Excludes the entity we are checking against.
+    $query->condition('mid', $entity->id(), '<>');
+    for ($i = 1; $i <= self::RESIZE_DIMENSION; $i++) {
+
+      // Calculate the number of color values that are considered "similar"
+      // given the percent threshold.
+      $color_difference = 100 / 255 * self::PIXEL_TOLLERANCE;
+
+      // Add conditions to the query for row and columns.
+      $query->condition('column_' . $i, $averages['columns'][$i - 1] - $color_difference, '>=');
+      $query->condition('column_' . $i, $averages['columns'][$i - 1] + $color_difference, '<=');
+
+      $query->condition('row_' . $i, $averages['rows'][$i - 1] - $color_difference, '>=');
+      $query->condition('row_' . $i, $averages['rows'][$i - 1] + $color_difference, '<=');
+    }
+    $result = $query->execute();
+
+    $mids = [];
+    while ($mid = $result->fetchField()) {
+      $mids[] = $mid;
+    }
+
+    return Media::loadMultiple($mids);
+  }
+
+  /**
+   * Get the array of data from the provided image URI.
+   *
+   * @param string $uri
+   *   Image path.
+   *
+   * @return array|bool
+   *   Array of color data or false if its not an image.
+   */
+  public function getColorData($uri) {
+    static $color_data = [];
```

Comment:
Swap out with: https://api.drupal.org/api/drupal/core%21includes%21bootstrap.inc/function/drupal_static/8.6.x

Reply:
https://www.drupal.org/project/currency/issues/1973114

Reply:
well make up ur mind. haha

Reply:
It is a process. Welcome to the journey.